### PR TITLE
Protect: Add Threat History

### PR DESCRIPTION
**Pull request by nateweller**
## Proposed changes:

* https://github.com/Automattic/jetpack/pull/37903
* https://github.com/Automattic/jetpack/pull/37988
* https://github.com/Automattic/jetpack/pull/38325
* https://github.com/Automattic/jetpack/pull/38626
* https://github.com/Automattic/jetpack/pull/38656
* https://github.com/Automattic/jetpack/pull/38094
* https://github.com/Automattic/jetpack/pull/38637
* https://github.com/Automattic/jetpack/pull/38703

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/jetpack-scan-team/issues/1225

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Ensure all included PRs have been reviewed and approved.
* Smoke test the Jetpack Protect plugin, specifically the new threat history section.

### CODE REVIEW

**Comment by dkmyta** [comment]
Something just came to mind while reviewing this, as part of `Jetpack_Protect::plugin_deactivation` we `Status::delete_option();` should we also clear `Scan_History` options here? 

I don’t think we clear all Protect/WAF options as part of deactivation currently, so I don’t know if its entirely necessary, just seems logical here since we do the same for `Status` (`Scan_` and/or `Protect_`). Also realizing that this looks like it might only clear one or the other, so if you have both one will likely remain. 

**UPDATE:** Just doubled checked this and it actually appears that neither are currently deleted on plugin deactivation. Seems the method is triggered but we aren't accurately detecting `static::OPTION_NAME` and `static::OPTION_TIMESTAMP_NAME`, because its instead accessing the values we define for these in the `Status` class. This is making me wonder if some of these other `Status` methods that use static variables are performing similarly.

File: projects/plugins/protect/src/js/components/threats-list/index.jsx [hunk 1]
```diff
@@ -123,27 +130,32 @@ const ThreatsList = () => {
...
+									<Text className={ styles[ 'manual-scan' ] } variant="body-small">
+										{ createInterpolateElement(
+											__(
+												'If you have manually fixed any of the threats listed above, <manualScanLink>you can run a manual scan now</manualScanLink> or wait for Jetpack to scan your site later today.',
+												'jetpack-protect'
+											),
+											{
+												manualScanLink: <ScanButton variant="link" />,
+											}
+										) }
```
Comment (dkmyta) [inline projects/plugins/protect/src/js/components/threats-list/index.jsx:157]: Now that this uses the new `ScanButton` component, the `isLoading` state is applied when a scan is enqueued and the display is a little awkward:
Screen Shot on 2024-08-09 at 09-49-36

File: projects/plugins/protect/src/class-rest-controller.php [hunk 2]
```diff
@@ -409,4 +454,29 @@ public static function api_complete_onboarding_steps( $request ) {
...
+	 */
+	public static function api_clear_scan_history_cache() {
+		$cache_cleared = Scan_History::delete_option();
+
+		if ( ! $cache_cleared ) {
+			return new WP_REST_Response( 'An error occured while attempting to clear the Jetpack Scan history cache.', 500 );
+		}
+
+		return new WP_REST_Response( 'Jetpack Scan history cache cleared.' );
+	}
```
Comment (nateweller) [inline projects/plugins/protect/src/class-rest-controller.php]:
```suggestion
```

Remove unused code.

**Review by dkmyta — DISMISSED** [review]
Everything looks great and tests well!

The items I mention in the prior comments should be addressed in followups, but as they don't specifically relate to the changes introduced here, I've gone ahead approved this one!

**Review by dkmyta — APPROVED** [review]

**Pull request merged**